osu.Game/Graphics/Containers/ScalingContainer.cs

```diff
@@ -118,6 +120,9 @@ private void load(OsuConfigManager config)
 
             posY = config.GetBindable<float>(OsuSetting.ScalingPositionY);
             posY.ValueChanged += _ => updateSize();
+
+            safeAreaPadding = safeArea.SafeAreaPadding.GetBoundCopy();
+            safeAreaPadding.BindValueChanged(_ => updateSize());
```

Comment:
This can potentially run on the input thread, as the safe area is updated from there.

Reply:
probably, yes.